### PR TITLE
Add mae, cross-entropy, and binary cross-entropy loss fns

### PR DESCRIPTION
**Pull request by joelshepherd**
This PR adds some common loss functions to compliment the existing `mse` loss.

I have chosen to implement the loss functions as creator functions `(options) => (y, yhat) => number` to accommodate adding configuration options to these functions in the future.

For example, torch's [`cross_entropy`](https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html) function accepts `reduction: "mean" | "sum" | "none"` as a configuration option.
This could then be available to users as such:
```ts
// model.ts
export const loss = sm.loss.crossEntropy({ reduction: "mean" });
```

But if you have other plans I am more than happy to remove the pattern.

### CODE REVIEW

**Comment by bwasti** [comment]
looks great!
